src/DataObjects/QueuedJobDescriptor.php

```diff
@@ -284,6 +285,30 @@ public function getTitle()
         return $this->JobTitle;
     }
 
+    /**
+     * Return a string representation of the numeric JobType
+     * @return string
+     */
+    public function getJobTypeString()
+    {
+        $map = $this->getJobTypeValues();
+        return isset($map[$this->JobType]) ? $map[$this->JobType] : '(Unknown)';
+    }
+
+
+    /**
+     * Return a map of numeric JobType values to localisable string representations.
+     * @return array
+     */
+    public function getJobTypeValues()
+    {
+        return [
+            QueuedJob::IMMEDIATE => _t(QueuedJob::class . '.TYPE_IMMEDIATE', 'Immediate'),
+            QueuedJob::QUEUED => _t(QueuedJob::class . '.TYPE_QUEUED', 'Queued'),
+            QueuedJob::LARGE => _t(QueuedJob::class . '.TYPE_LARGE', 'Large'),
```

Comment:
Unfortunately `Some::class` doesn't work in the text collector. Will need to be `_t('Symbiote\\QueuedJobs\\Services\\QueuedJob',`

Reply:
Doh, you're right https://github.com/silverstripe/silverstripe-framework/issues/7647

I might just use `__class__`; these values will be attached to QueuedJobDescriptor but that's not a showstopper?

Reply:
Yep sounds good

Reply:
Done; squashed in.